src/com/company/Main.java

```diff
@@ -1,8 +1,151 @@
-	// write your code here
+        // write your code here
+        //task1
+        isYearLeap(2020);
+
+        //task2
+        int currentYear = LocalDate.now().getYear();
+        displayAppVersionToInstall(1,currentYear);
+
+        //task3
+        daysToDeliverCard(122);
+
+        //task4
+        validateString("aabccddefgghiijjkk");
+
+        //task5
+        int[] initialArray = new int[]{3, 2, 1, 6, 5};
+        makeReversiveArray(initialArray);
+        System.out.println("Развернутый согласно Заданию 5 массив выглядит так: "+Arrays.toString(initialArray));
+
+//
+////  example1
+//        int num1 = 18;
+//        int num2 = 8;
+//        int[] arr = {3, 4, 5};
+//    printGreetings("George");
+//    calculateSum(num1,num2);
+//    calculateSum(arr);
+//
+////  example2
+//        int a2 = 1;
+//        int[] arr2 = {1, 2, 3};
+//        changeValues(a2,arr2);
+//        System.out.println(a2);
+//        System.out.println(Arrays.toString(arr2));
+//    }
+//
+////  shpora_example1
+//    public static void calculateSum(int[] arr) {
+//        int sum = 0;
+//        for (int i : arr) {
+//            sum += i;
+//        }
+//        System.out.println("Сумма элементов массива: " + sum);
+//    }
+////  shpora_example2
+//    public static void calculateSum(int a, int b) {
+//        int sum = a + b;
+//        System.out.println("Сумма элементов: " + sum);
+//    }
+////  shpora_example3
+//    public static void printGreetings(String name) {
+//        System.out.println("Привет, " + name);
+//    }
+////  shpora_example4
+////    public static void changeValues(int a2, int [] arr3){
+////        a2 = 5;
+////        arr3[0] = 5;
+////    }
+//
+//    //  shpora_example5
+//    public static void changeValues(int a2, int[] arr3) {
+//        a2 = 5;
+//        arr3 = new int[10];
+//        arr3[0] = 5;
+    }
+
+    //      task1|method
+    public static void isYearLeap(int year){
+        if (year % 400 == 0) {
+            System.out.println(year + "г = високосный год");
+        } else if (year % 100 == 0) {
+            System.out.println(year + "г = не високосный год");
+        } else if (year % 4 == 0) {
+            System.out.println(year + "г = високосный год");
+        } else {
+            System.out.println(year + "г = не високосный год");
+        }
+    }
+
+    //      task2|method
+    public static void displayAppVersionToInstall(int clientOS, int clientDeviceYear) {
+        if (clientDeviceYear < 2015)
+            switch (clientOS) {
+                case 0:
+                    System.out.println("Установите облегченную версию приложения для iOS по ссылке");
+                    break;
+                case 1:
+                    System.out.println("Установите облегченную версию приложения для Android по ссылке");
+                    break;
+                default:
+                    System.out.println("Выберите ссылку для скачивания облегченной версии приложения вручную");
+            }
```

Comment:
можно заменить на улучшенный switch
```suggestion
              switch (clientOS) {
                case 0 -> System.out.println("Установите облегченную версию приложения для iOS по ссылке");
                case 1 -> System.out.println("Установите облегченную версию приложения для Android по ссылке");
                default -> System.out.println("Выберите ссылку для скачивания облегченной версии приложения вручную");
            }
```

Reply:
а, точно, забыл, что так можно) спс за напоминание!)